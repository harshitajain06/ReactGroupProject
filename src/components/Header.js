import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import logoImg from '../images/logo.png';
import '../styles/header.css';

const Header = () => {
  const [link, setLink] = useState('');
  const styles = {
    padding: '0 15px',
    color: '#0290ff',
  };
  const handleClick = (e) => {
    setLink(e.target.classList.toggle('active-link'));
  };
  return (
    <div>
      <nav>
        <div className="logo-title">
          <img className="logo-img" src={logoImg} alt="logo" />
          <h1 className="title">Space Travelers&apos; Hub</h1>
        </div>
        <div className="links">
          <Link to="/" style={styles} className="link" onClick={handleClick}>
            Rockets
            {' '}
            {link}
          </Link>
          <Link
            to="/missions"
            style={styles}
            className="link"
            onClick={handleClick}
          >
            Missions
          </Link>
          |
          <Link
            to="/myprofile"
            style={styles}
            className="link"
            onClick={handleClick}
          >
            My Profile
          </Link>
        </div>
      </nav>
      <hr />
    </div>
  );
};

export default Header;
