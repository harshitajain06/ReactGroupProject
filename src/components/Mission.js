import React from "react";
import "../styles/mission.css";

const Mission = (props) => {
  const { missionName, desc } = props;
  return (
    <div className="mission">
      <h4 className="mission-name">{missionName}</h4>
      <p className="mission-desc">{desc}</p>
      <div className="mission-btn mission-status">
        <button type="button" className="status-button">
          NOT A MEMBER
        </button>
      </div>
      <div className="mission-btn mission-join">
        <button type="button" className="join">
          JOIN MISSION
        </button>
      </div>
    </div>
  );
};

export default Mission;
