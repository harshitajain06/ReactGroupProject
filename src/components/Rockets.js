import { useDispatch, useSelector } from 'react-redux';
import { cancelReservation, reserveRocket } from '../redux/rockets/rockets';
import '../styles/rockets.css';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();
  return (
    <div>
      <ul className="rocket-ul">
        {rockets.map((r) => (
          <li key={r.id} className="rocket-li">
            <img className="rocket-img" src={r.flickr_images} alt={r.id} />
            <div className="rocket-txt">
              <h2>{r.name}</h2>
              {r.reserved === true ? <span className="reserveBtn">&bull; Reserved</span> : ''}
              <p>{r.description}</p>
              {r.reserved === false ? <button className="btn1" onClick={() => { dispatch(reserveRocket(r.id)); }} type="button">MAKE RESERVATION</button> : <button className="btn2" onClick={() => { dispatch(cancelReservation(r.id)); }} type="button">CANCEL RESERVATION</button>}

            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Rockets;
