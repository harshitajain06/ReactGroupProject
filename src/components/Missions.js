import React, { useEffect, useRef } from "react";
import { useDispatch, useSelector } from "react-redux";
import Mission from "./Mission";
import "../styles/missions.css";
import { getMission } from "../redux/missions/missions";

const Missions = () => {
  const shouldGetMissions = useRef(true);
  const dispatch = useDispatch();
  useEffect(() => {
    if (shouldGetMissions.current) {
      shouldGetMissions.current = false;
      dispatch(getMission());
    }
    // eslint-disable-next-line
  }, []);

  const missions = useSelector((state) => state.missions);

  return (
    <div className="missions-page">
      <div className="content">
        <h4 className="content-missions">Missions</h4>
        <h4 className="content-desc">Description</h4>
        <h4 className="content-status">Status</h4>
        <h4 className="content-join">Join/Leave</h4>
      </div>
      <div className="missions">
        {missions.map((mission) => (
          <Mission
            key={mission.mission_id}
            missionName={mission.mission_name}
            desc={mission.description}
            id={mission.mission_id}
            status={mission.status}
            joined={mission.joined}
          />
        ))}
      </div>
    </div>
  );
};

export default Missions;
