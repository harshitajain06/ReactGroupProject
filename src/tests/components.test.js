import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../redux/configureStore";
import Mission from "../components/Mission";
import Missions from "../components/Missions";
import Header from "../components/Header";
import Rockets from "../components/Rockets";

describe("Test components", () => {
  it("Mission component renders correctly", () => {
    const mission = render(
      <Provider store={store}>
        <Mission />
      </Provider>
    );

    expect(mission).toMatchSnapshot();
  });

  it("Missions component renders correctly", () => {
    const missions = render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );

    expect(missions).toMatchSnapshot();
  });

  it("Rockets component renders correctly", () => {
    const rockets = render(
      <Provider store={store}>
        <Rockets />
      </Provider>
    );
    expect(rockets).toMatchSnapshot();
  });

  it("MyProfile component renders correctly", () => {
    const myProfile = renderer.create(<myProfile />).toJSON();
    expect(myProfile).toMatchSnapshot();
  });
});

// Test Navigation links
const MockNavLinks = () => {
  <BrowserRouter>
    <Header />
  </BrowserRouter>;
};
describe("Test links in Header component", () => {
  it("Nav links render correctly", () => {
    const navLinks = renderer.create(<MockNavLinks />).toJSON();
    expect(navLinks).toMatchSnapshot();
  });
});
