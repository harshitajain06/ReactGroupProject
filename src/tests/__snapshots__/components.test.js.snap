// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components Mission component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mission"
      >
        <h4
          class="mission-name"
        >
           
        </h4>
        <p
          class="mission-desc"
        />
        <div
          class="member-status"
        >
          <p
            class="inactive-member"
          >
            Not a member
          </p>
        </div>
        <div
          class="join-cancel-btn"
        >
          <button
            class="join-mission"
            type="button"
          >
            Join Mission
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mission"
    >
      <h4
        class="mission-name"
      >
         
      </h4>
      <p
        class="mission-desc"
      />
      <div
        class="member-status"
      >
        <p
          class="inactive-member"
        >
          Not a member
        </p>
      </div>
      <div
        class="join-cancel-btn"
      >
        <button
          class="join-mission"
          type="button"
        >
          Join Mission
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test components Missions component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions-page"
      >
        <div
          class="content"
        >
          <h4
            class="content-mission"
          >
            Mission
          </h4>
          <h4
            class="content-desc"
          >
            Description
          </h4>
          <h4
            class="content-status"
          >
            Status
          </h4>
          <h4
            class="content-join"
          >
            Join/Leave
          </h4>
        </div>
        <div
          class="missions"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions-page"
    >
      <div
        class="content"
      >
        <h4
          class="content-mission"
        >
          Mission
        </h4>
        <h4
          class="content-desc"
        >
          Description
        </h4>
        <h4
          class="content-status"
        >
          Status
        </h4>
        <h4
          class="content-join"
        >
          Join/Leave
        </h4>
      </div>
      <div
        class="missions"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test components MyProfile component renders correctly 1`] = `<myProfile />`;

exports[`Test components Rockets component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <ul
          class="rocket-ul"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="rocket-ul"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test links in Header component Nav links render correctly 1`] = `null`;
